bundle/jsky.app.ot/src/main/java/jsky/app/ot/tpe/feat/TpeTargetPosFeature.java

```diff
@@ -26,11 +27,59 @@
         _props.registerBooleanProperty(PROP_SHOW_TAGS, true);
     }
 
+    private final class UserCreateableItem implements TpeCreatableItem {
+        private final UserTarget.Type userTargetType;
+
+        UserCreateableItem(UserTarget.Type t) {
+            this.userTargetType = t;
+        }
+
+        public String getLabel() {
+            return userTargetType.displayName;
+        }
+
+        public Type getType() {
+            return Type.userTarget;
+        }
+
+        public boolean isEnabled(TpeContext ctx) {
+            return ctx.targets().isDefined();
+        }
+
+        public void create(TpeMouseEvent tme, TpeImageInfo tii) {
+            final TargetObsComp obsComp = getTargetObsComp();
+            if (obsComp == null) return;
+
+            final Option<SiderealTarget> targetOpt = tme.target;
+            final SPTarget userPos = targetOpt.map(SPTarget::new).getOrElse(() -> {
+                final double ra  = tme.pos.ra().toDegrees();
+                final double dec = tme.pos.dec().toDegrees();
+                // No SkyObject info is present so we use the old way of creating
+                // a target from a mouse event using only the coordinates.
+                return new SPTarget(ra, dec);
```

Comment:
Only out of the `Supplier` implementation.  It needs a `return` if I understand it correctly because it is multiple lines.

I didn't pay much attention to this code, which I moved up from lower down in the original source.  I did add the `UserTarget.Type` parameter to the constructor  and have it make appropriately typed user targets.  Now that I'm looking at it, I see that it is deferring the creation of the new `SPTarget` to when/if it is necessary by using a `java.util.function.Supplier`.